library/std/src/process.rs

```diff
@@ -558,6 +558,12 @@ impl Command {
     /// but this has some implementation limitations on Windows
     /// (see issue #37519).
     ///
+    /// Note on Windows: For executable files with the .exe extension,
+    /// it can be omitted when specifying the program for this Command.
+    /// However, for other types of executable files like .bat or .cmd,
+    /// a filename including the extension needs to be provided,
+    /// otherwise the file won't be found.
```

Comment:
How about something like:
```suggestion
    /// However, if the file has a different extension
    /// a filename including the extension needs to be provided,
    /// otherwise the file won't be found.
```

Ultimately I think we need a lot more documentation on the way file names are resolved and run, including caveats, but that's perhaps a bigger job than this PR. And until we do I'm nervous about actively encouraging people to use `.bat` files in `Command::new` because it was an accidental feature that we've just tried not to break too badly.